invenio_app_ils/ui/src/invenio_app_ils/common/components/ESSelector/ESSelectorModal.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal } from 'semantic-ui-react';
+import { ESSelector } from './';
+import './ESSelector.scss';
+
+export default class ESSelectorModal extends Component {
+  state = { visible: false };
+
+  constructor(props) {
+    super(props);
+    this.selectorRef = null;
+  }
+
+  show = () => this.setState({ visible: true });
+
+  hide = () => this.setState({ visible: false });
+
+  toggle = () => this.setState({ visible: !this.state.visible });
+
+  save = () => {
+    const { multiple, onSave } = this.props;
+    if (onSave) {
+      if (multiple) {
```

Comment:
what does the multiple objects to save change?

Reply:
The idea was that if you enable multi-selection the results returned to `onSave` will be an array with the selections. If you only allow a single selection it made more sense to return the selection as-is and not in an array of one item. But it might be confusing and perhaps it's better to always return an array even if only a single selection is allowed?

Reply:
I would simplify it to have always the same structure, but just block in single mode possibility of adding more, like you proposed